src/Typeahead.react.js

```diff
@@ -535,10 +541,12 @@ const Typeahead = React.createClass({
         break;
       case RETURN:
         // Prevent submitting forms.
-        e.preventDefault();
-
-        if (showMenu) {
-          activeItem && this._handleAddOption(activeItem);
+        if (showMenu && activeItem) {
```

Comment:
oh, definitely, you are right it I have messed but I think it should be something like this:

```javascript
if (!this.props.formSubmit || showMenu && activeItem) {
  e.preventDefault();
}

if (showMenu && activeItem) {
  this._handleAddOption(activeItem);
}
```

Because we don't want to submit form when user just selected menu item.